### PR TITLE
[PF-1613], [PF-1883]  Fix for RBS swagger-ui page, Javadoc warnings

### PR DESCRIPTION
**Pull request by dexamundsen**
[PF-1883]  - Update comments to resolve Javadoc warnings

[PF-1613]
- Fix broken swagger UI on RBS
- Minor clean up
- Local testing screenshot below
image

[PF-1883]: https://broadworkbench.atlassian.net/browse/PF-1883?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PF-1613]: https://broadworkbench.atlassian.net/browse/PF-1613?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/bio/terra/buffer/db/BufferDao.java [hunk 1]
```diff
@@ -544,6 +543,7 @@ private static CloudResourceUid deserializeResourceUid(String cloudResourceUid)
     }
   }
 
+  // TODO: move to terra-common-lib PF-1912
   /** Helper function to return Millisecond precision instant supported by most DBs */
   public static Instant currentInstant() {
```
Comment (ddietterich) [inline src/main/java/bio/terra/buffer/db/BufferDao.java:548]: I don't understand what this is needed for. Do you?

I have not seen this concern in any other code. In WSM we move easily from Instant to SQL Timestamp types without concern for the nanoseconds.

**Review by ddietterich — APPROVED** [review]
Looks like a good cleanup. What change fixed the swagger page? Just renaming the file?

File: src/main/java/bio/terra/buffer/app/controller/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -1,19 +1,58 @@
...
  * This module provides a top-level exception handler for controllers. All exceptions that rise
  * through the controllers are caught in this handler. It converts the exceptions into standard
  * ErrorReport responses.
  */
 @RestControllerAdvice
-public class GlobalExceptionHandler extends AbstractGlobalExceptionHandler<ErrorReport> {
-  public ErrorReport generateErrorReport(Throwable ex, HttpStatus statusCode, List<String> causes) {
-    return new ErrorReport().message(ex.getMessage()).statusCode(statusCode.value()).causes(causes);
+public class GlobalExceptionHandler {
+  // TODO: move to terra-common-lib PF-1912
```
Comment (yuhuyoyo) [inline src/main/java/bio/terra/buffer/app/controller/GlobalExceptionHandler.java:22]: i don't understand what to be moved to TCL. the logger? 

And a nit, could you format the TODO as TODO(PF-1912): <action>

also just curious, is this to fix the RBS swagger-ui?

File: src/main/resources/static/service_openapi.yaml [hunk 3]
```diff
@@ -111,8 +111,8 @@ components:
       type: object
```
Comment (yuhuyoyo) [inline src/main/resources/static/service_openapi.yaml:111]: is changing the folder from api -> static also for fixing RBS swagger ui?

File: .github/workflows/master_push.yml [hunk 4]
```diff
@@ -3,7 +3,7 @@ name: Tag, Build, and Push Image
 on:
   push:
     branches: [ master ]
-    paths-ignore: [ '**.md',  'LICENSE.txt' ]
```
Comment (yuhuyoyo) [inline .github/workflows/master_push.yml:6]: unrelated to the tickets?

**Review by zloery — APPROVED** [review]

**Comment by dexamundsen** [comment]
> Looks like a good cleanup. What change fixed the swagger page? Just renaming the file?

Yep - changing the parent folder name to static fixed the error

File: src/main/java/bio/terra/buffer/db/BufferDao.java (hunk 1 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/buffer/db/BufferDao.java:548]: @ddietterich 
 Here is the issue - 
- Unit tests run on local machine with DB on docker  -> no errors. 
- Unit tests run as part of PR consistently failed on some of the test cases: comparing expected objects to those inserted/selected from DB (or other such conversions) errored out on precision. One such example - https://github.com/DataBiosphere/terra-resource-buffer/runs/7832530360?check_suite_focus=true#step:14:83 (unit test: line82)

Picked MILLISECOND precision over micro

File: src/main/java/bio/terra/buffer/app/controller/GlobalExceptionHandler.java (hunk 2 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/buffer/app/controller/GlobalExceptionHandler.java:22]: This exception handler mechanism is the same and used across WSM, RBS, Janitor, etc.. when swagger UI is involved. We could reduce code repetition by adding an error handling mechanism part of a library / shared codebase. 
Not sure if TCL is the best place, was just an initial thought. 
note - this is added as a comment in Janitor code and i copied from that source

Comment (dexamundsen) [reply src/main/java/bio/terra/buffer/app/controller/GlobalExceptionHandler.java:22]: Done, updated the TODO format

File: .github/workflows/master_push.yml (hunk 4 above)
Comment (dexamundsen) [reply .github/workflows/master_push.yml:6]: Added to ticket description, and piggybacking here since it is a minor change

**Pull request merged**